### PR TITLE
Hover-highlight for link to a section in a post shows excerpt starting at that section

### PR DESCRIPTION
**Pull request by jimrandomh**
This makes it so that if a link to a post includes the anchor of a section heading, the excerpt that you see when you hover over it starts at that section, rather than at the top of the post. Partially addresses https://github.com/LessWrong2/Lesswrong2/issues/4072 . Current limitations:

 * Currently only section headings have anchors (the anchors are inserted by the table of contents code), so you can only point this at sections that start with a large-font header, not at arbitrary paragraphs
 * You'll get a brief loading spinner while it roundtrips to the server, before seeing the excerpt
 * The excerpt is clipped by length in the same way as an excerpt from the start of the post, so if the section is short enough, it'll be followed by the start of the next section

### CODE REVIEW

**Comment by jimrandomh** [comment]
Fixes #3891 .

**Comment by darkruby501** [comment]
I've had a quick look at this but there were multiple elements I wasn't familiar with. I was hoping to go through it with Oli, of course Oli is hard to get ahold. @jimrandomh Maybe you're up for walking me through it on a call?

File: packages/lesswrong/components/posts/PostsPreviewTooltip.tsx [hunk 1]
```diff
@@ -139,20 +140,31 @@ const getPostCategory = (post: PostsBase) => {
...
-  const { PostsUserAndCoauthors, PostsTitle, ContentItemBody, CommentsNode, BookmarkButton, LWTooltip, FormatDate } = Components
+  const { PostsUserAndCoauthors, PostsTitle, ContentItemBody, CommentsNode, BookmarkButton, LWTooltip, FormatDate, Loading } = Components
   const [expanded, setExpanded] = useState(false)
 
-  if (!post) return null
+  const {document: postWithHighlight, loading} = useSingle({
+    collectionName: "Posts",
+    fragmentName: "HighlightWithHash",
+    documentId: post?._id,
+    skip: !post || !hash,
```
Comment (Discordius) [inline packages/lesswrong/components/posts/PostsPreviewTooltip.tsx:157]: If I understand this correctly, this now means all hover-previews for posts have an initial loading screen. I don't think this is worth the tradeoff. I feel like we should just keep using the `post` element that is passed if we are not being passed a `hash`, which skips the loading, and only load the postWithHighlight if we have indeed been passed a `hash` and that way we don't have any regression in user experience.

**Review by Discordius — CHANGES_REQUESTED** [review]

File: packages/lesswrong/components/posts/PostsPreviewTooltip.tsx (hunk 1 above)
Comment (jimrandomh) [reply packages/lesswrong/components/posts/PostsPreviewTooltip.tsx:157]: You misunderstood; this won't have a loading spinner if there's no `hash` (I just tested).

Comment (Discordius) [reply packages/lesswrong/components/posts/PostsPreviewTooltip.tsx:157]: Cool, then all seems good. Somewhat surprised by this, but will check it out in prod.

**Pull request merged**